src/esp/metadata/MetadataMediator.cpp

```diff
@@ -36,39 +117,271 @@ bool MetadataMediator::createDataset(const std::string& sceneDatasetName,
+                << " does not reference a SceneAttributes named  " << sceneName
+                << " but a SceneAttributes config named "
+                << sceneFilenameCandidate
+                << " was found on disk, so "
+                   "loading.";
+      sceneAttributes = dsSceneAttrMgr->createObjectFromJSONFile(
+          sceneFilenameCandidate, true);
+    } else {
+      // get list of stage attributes handles that contain sceneName as a
+      // substring
+      auto stageList = dsStageAttrMgr->getObjectHandlesBySubstring(sceneName);
+      if (stageList.size() > 0) {
+        // 3.  Existing, registered StageAttributes in current active dataset.
+        //    In this case, a SceneAttributes is created amd registered using
+        //    sceneName, referencing the StageAttributes of the same name; This
+        //    sceneAttributes is returned.
+        LOG(INFO) << "MetadataMediator::getSceneAttributesByName : No existing "
+                     "scene instance attributes containing name "
+                  << sceneName << " found in Dataset : " << activeSceneDataset_
+                  << " but " << stageList.size()
+                  << " StageAttributes found.  Using " << stageList[0]
+                  << " as stage and to construct a SceneAttributes with same "
+                     "name that will be added to Dataset.";
+        // create a new SceneAttributes, and give it a
+        // SceneObjectInstanceAttributes for the stage with the same name.
+        sceneAttributes = makeSceneAndReferenceStage(
+            datasetAttr, dsStageAttrMgr->getObjectByHandle(stageList[0]),
+            dsSceneAttrMgr, sceneName);
+
+      } else {
+        // 4.  Existing stage config/asset on disk, but not in current dataset.
+        //    In this case, a stage attributes is loaded and registered, then
+        //    added to current dataset, and then 3. is performed.
+        LOG(INFO)
+            << "MetadataMediator::getSceneAttributesByName : Dataset : "
+            << activeSceneDataset_
+            << " has no preloaded SceneAttributes or StageAttributes named : "
+            << sceneName
+            << " so loading/creating a new StageAttributes with this "
+               "name, and then creating a SceneAttributes with the same name "
+               "that referneces this stage.";
+        // create and register stage
+        auto stageAttributes = dsStageAttrMgr->createObject(sceneName, true);
+        // create a new SceneAttributes, and give it a
+        // SceneObjectInstanceAttributes for the stage with the same name.
+        sceneAttributes = makeSceneAndReferenceStage(
+            datasetAttr, stageAttributes, dsSceneAttrMgr, sceneName);
+      }
+    }
+  }
+  // make sure that all stage, object and lighting attributes referenced in
+  // scene attributes are loaded in dataset, as well as the scene attributes
+  // itself.
+  datasetAttr->addNewSceneInstanceToDataset(sceneAttributes);
```

Comment:
That's what the function does.  If it does not find named attributes within the dataset corresponding to those specified in the scene instance, it attempts to create them.   This process will not "fail" per se, but empty results/unfound attributes may not be what the user wishes for.  Or, it may be, depending on the source of the scene instance attributes that has been created.